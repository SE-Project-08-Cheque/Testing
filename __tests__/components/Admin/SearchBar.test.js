// import dependencies
import React from "react";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import react-testing methods
import {  cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import the component for testing
import SearchBar from "../../../components/Admin/SearchBar";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <SearchBar/>,
    div
  );
});

it("matches snapshot", () => {
  const tree = renderer.create(
    <SearchBar/>
  );

  expect(tree).toMatchSnapshot();
});
