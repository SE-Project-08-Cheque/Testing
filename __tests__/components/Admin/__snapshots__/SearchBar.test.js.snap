// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-1ws0jr4"
>
  <div
    className="chakra-input__group css-4302v8"
  >
    <div
      className="chakra-input__right-element css-a6fysx"
    >
      <button
        aria-label="Search database"
        className="chakra-button css-1kgd4fd"
        type="button"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-onkibi"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M23.384,21.619,16.855,15.09a9.284,9.284,0,1,0-1.768,1.768l6.529,6.529a1.266,1.266,0,0,0,1.768,0A1.251,1.251,0,0,0,23.384,21.619ZM2.75,9.5a6.75,6.75,0,1,1,6.75,6.75A6.758,6.758,0,0,1,2.75,9.5Z"
            fill="currentColor"
          />
        </svg>
      </button>
    </div>
    <input
      className="chakra-input css-0"
      name="search"
      onBlur={[Function]}
      onFocus={[Function]}
    />
  </div>
</div>
`;
